### PR TITLE
Feature/markdown implementation

### PR DESCRIPTION
**Pull request by DanielAlferezCode**
Requirement: [Markdown parsing](https://coda.io/d/_dlsKLnttMd2#REQUIREMENTS_tugRQZMI/r1221&view=center)

## Summary by CodeRabbit

- **New Features**
  - End-to-end Markdown: client renderer + preview and a POST endpoint to convert Markdown to sanitized HTML.
  - Improved AI response rendering: supports tables, code blocks, and charts in messages.
  - External links default to opening in a new tab with security attributes.

- **Style**
  - Global Markdown styles added; thinner separators and standardized chart typography.
  - Code block copy control moved to a sticky, right-aligned layout.

- **Tests**
  - Extensive unit tests and deterministic mocks covering API, rendering, sanitization, edge cases, and failures.

- **Chores**
  - Test tooling and dependencies updated to support Markdown parsing and sanitization.

### CODE REVIEW

File: __tests__/components/AnalysisChat/AIResponseBubble.test.tsx [hunk 1]
```diff
@@ -45,9 +104,10 @@ describe('AIResponseBubble', () => {
...
       screen.getByText('Formatted(2025-07-23T10:00:00.000Z)')
     ).toBeInTheDocument();
-    expect(
-      screen.getByText('This is an AI response message.')
-    ).toBeInTheDocument();
+    const contentEl = await screen.findByText(
+      'This is an AI response message.'
+    );
+    expect(contentEl).toBeInTheDocument();
   });
```
[the remark this answers is not in the document]
Comment (DanielAlferezCode) [reply __tests__/components/AnalysisChat/AIResponseBubble.test.tsx:111]: I updated the test to be async and kept findByText (Option A). Biome stops flagging it and tests pass locally. I also checked for similar cases in this suite and didn’t find any others.

File: src/components/Markdown/MarkdownPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  html: SafeHtml;
+  className?: string;
+}>;
+
+export default function MarkdownPreview({
+  html,
+  className,
+}: MarkdownPreviewProps) {
+  return (
+    <div className={className} dangerouslySetInnerHTML={{ __html: html }} />
```
[the remark this answers is not in the document]
Comment (DanielAlferezCode) [reply src/components/Markdown/MarkdownPreview.tsx:17]: I’ve added an inline comment explaining why dangerouslySetInnerHTML is safe here, without referencing Biome directly. The note clarifies that the html prop is already sanitized via DOMPurify and branded as SafeHtml.

File: src/lib/utils/markdown.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      }
+    );
+
+    // Enforce anchor target/rel
+    purifier.addHook("afterSanitizeAttributes", (node: Element) => {
+      if (node.tagName === "A" && node.getAttribute("href")) {
+        node.setAttribute("target", "_blank");
+        node.setAttribute("rel", "noopener noreferrer");
+      }
+    });
```
[the remark this answers is not in the document]
Comment (DanielAlferezCode) [reply src/lib/utils/markdown.ts]: I’ll keep the current implementation — it already limits target="_blank" to external links, preserves existing rel values while adding the required security attributes, and also cleans up _blank from non-external links. This approach is safer and more flexible than removing attributes unconditionally.
